### PR TITLE
MSGID_REQUEST_FAILURE to response MSGID_GLOBAL_REQUEST

### PR DESCRIPTION
**Pull request by kojo1**
Response SSH_MSG_REQUEST_FAILURE to Keep-Alive. IETF:draft-ssh-global-requests

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -7579,6 +7601,38 @@ int SendRequestSuccess(WOLFSSH* ssh, int success)
...
+    int ret = WS_SUCCESS;
+
+    WLOG(WS_LOG_DEBUG, "Entering SendRequestFailure");
+
+    if (ssh == NULL)
+        ret = WS_BAD_ARGUMENT;
+
+    if (ret == WS_SUCCESS)
+        ret = PreparePacket(ssh, MSG_ID_SZ);
+    printf("BundlePacket, ret=%d\n", ret);
```
Comment (ejohnstown) [inline src/internal.c]: Debug print that should be removed.

File: src/internal.c [hunk 2]
```diff
@@ -7579,6 +7601,38 @@ int SendRequestSuccess(WOLFSSH* ssh, int success)
...
+    if (ret == WS_SUCCESS)
+    {
+        output = ssh->outputBuffer.buffer;
+        idx = ssh->outputBuffer.length;
+
+        output[idx++] = MSGID_REQUEST_FAILURE;
+
+        ssh->outputBuffer.length = idx;
+        printf("BundlePacket\n");
```
Comment (ejohnstown) [inline src/internal.c]: Another debug print.

File: src/internal.c [hunk 3]
```diff
@@ -7569,71 +7569,40 @@ int SendUserAuthBanner(WOLFSSH* ssh)
...
-
-    if (ret == WS_SUCCESS)
-        ret = wolfSSH_SendPacket(ssh);
-
-    WLOG(WS_LOG_DEBUG, "Leaving SendRequestSuccess(), ret = %d", ret);
-    return ret;
-}
-
-int SendRequestFailure(WOLFSSH *ssh)
+int SendRequestSuccess(WOLFSSH *ssh, int success)
```
Comment (ejohnstown) [inline src/internal.c]: Can we treat the success parameter as a boolean? We are currently using it with an error code enumeration (WS_SUCCESS) that is getting complemented (~WS_SUCCESS).

File: src/internal.c [hunk 4]
```diff
@@ -3869,7 +3869,7 @@ static int DoGlobalRequest(WOLFSSH* ssh,
         *idx += len;
 
         if (wantReply)
-            ret = SendRequestFailure(ssh);
+            ret = SendRequestSuccess(ssh, ~WS_SUCCESS);
```
Comment (ejohnstown) [inline src/internal.c]: If the second parameter is treated as boolean, ~WS_SUCCESS can just be 0. WS_SUCCESS is supposed to be a named return code, not treated as a number.

File: src/internal.c (hunk 3 above)
Comment (kojo1) [reply src/internal.c]: Done

File: src/internal.c (hunk 4 above)
Comment (kojo1) [reply src/internal.c]: Done

**Comment by kojo1** [comment]
Use needs is just for the failure for now. Put both of them,  just in case.

**Pull request merged**